File: roam/py/distributions.py
# :PROPERTIES:
# :ID:       42923711-fe51-4122-9c42-5f8887afe801
# :header-args: :tangle py/distributions.py :comments both
# :END:
# #+title: py/util/distributions

# 分布工具


# [[file:../20250517115402-py_util_dist_utils.org::+BEGIN_SRC python][No heading:1]]
import random
# No heading:1 ends here

# [[file:../20250517115402-py_util_dist_utils.org::+BEGIN_SRC python][No heading:2]]
def avange(a, b, size):
    return [random.uniform(a, b) for _ in range(size)]
# No heading:2 ends here

# [[file:../20250517115402-py_util_dist_utils.org::+BEGIN_SRC python][No heading:3]]
def gauss(size):
    return [random.gauss(0, 1) for _ in range(size)]
# No heading:3 ends here

File: roam/hs/Distributions.hs
-- :PROPERTIES:
-- :ID:       1433c0e7-9496-4a8a-a217-f0dd5c7ff50c
-- :header-args: :tangle hs/Distributions.hs :comments both
-- :END:
-- #+title: hs/util/distributions


-- Haskell 没有内置 高斯分布生成器，我们得自己实现.


-- [[file:../20250519161401-hs_util_distributions.org::+BEGIN_SRC haskell][No heading:1]]

-- No heading:1 ends here

import Graphics.Rendering.OpenGL
import Graphics.GLU

main :: IO ()
main = do
  (_progName, _args) <- getArgsAndInitialize
  _window <- createWindow "OpenGL Demo"

display = do
  clear [ColorBuffer]
  renderPrimitive Triangles $ do
    vertex $ Vertex2 (-0.5) (-0.5)
    vertex $ Vertex2 0.5 (-0.5)
    vertex $ Vertex2 0 0.5
